### PR TITLE
vimPlugins.idris2-nvim: init at 2023-09-05

### PR DESCRIPTION
**Pull request by mattpolzin**
## Description of changes

Adds the [`idris2-nvim`](https://github.com/ShinKage/idris2-nvim) plugin. This similarly named plugin is distinct from `idris-vim` (a plugin for Idris 1) and `idris2-vim` (an older plugin for Idris2). This newer plugin uses the Idris 2 language server.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by mattpolzin** [comment]
Result of `nixpkgs-review` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vimPlugins.idris2-nvim</li>
  </ul>
</details>

**Comment by mattpolzin** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vimPlugins.idris2-nvim</li>
  </ul>
</details>

**Comment by mattpolzin** [comment]
@GaetanLepage would you be interested in reviewing?

**Comment by GaetanLepage** [comment]
> @GaetanLepage would you be interested in reviewing?

Sure, this looks good to me !

**Review by GaetanLepage — APPROVED** [review]

File: pkgs/applications/editors/vim/plugins/overrides.nix [hunk 1]
```diff
@@ -723,6 +723,10 @@
     src = "${hurl.src}/contrib/vim";
   };
 
+  idris2-nvim = super.idris2-nvim.overrideAttrs {
+    dependencies = with self; [ nui-nvim nvim-lspconfig ];
+  };
```
Comment (teto) [inline pkgs/applications/editors/vim/plugins/overrides.nix:728]: could you add 
```
    doInstallCheck = true;
    nvimRequireCheck = "idris2";
```
to check we are not missing anything ? also nui-nvim is said to be optional so I would probably remove it

Comment (mattpolzin) [reply pkgs/applications/editors/vim/plugins/overrides.nix:728]: Happy to add those lines to the override. I'll sanity check first, but my recollection is that the plugin fails to start without nui; I assume the README is out of date about it being optional. If I find things do start up without nui I'll remove it!

Comment (teto) [reply pkgs/applications/editors/vim/plugins/overrides.nix:728]: hehe that's why I asked. My goal is to run more and more upstream tests for neovim plugins. 
In absence of those or as a quick hack, the `nvimRequireCheck` is hepful  and already  helped find missing depenencies like that.

**Comment by teto** [comment]
ty for the contribution

**Pull request merged**